devops/scripts/register_bundle_with_api.sh

```diff
@@ -109,64 +100,64 @@ explain_json=$(porter explain --reference "${acr_name}".azurecr.io/"$(yq eval '.
-  then
-    echo "API access token isn't available - automatic bundle registration not possible. Use the script output to self-register. "
+if [ "${dry_run}" == "true" ]; then
+    echo "--dry-run specified - automatic bundle registration disabled. Use the script output to self-register. "
     echo "See documentation for more details: https://microsoft.github.io/AzureTRE/tre-admins/registering-templates/"
     echo "${payload}" | jq --color-output .
     exit 1
-  fi
-  access_token=${ACCESS_TOKEN}
 fi
+
 if [ "${bundle_type}" == "user_resource" ] && [ -z "${workspace_service_name:-}" ]; then
   echo -e "You must supply a workspace service_name name if you would like to automatically register the user_resource bundle\n"
   echo "${payload}" | jq --color-output .
   usage
 fi
 
-if [[ -n ${insecure+x} ]]; then
-    options="-k"
-fi
+template_name=$(yq eval '.name' porter.yaml)
+template_version=$(yq eval '.version' porter.yaml)
 
-if [[ -z ${tre_url} ]]; then
-  # access_token specified but no URL
-  echo -e "No TRE URL provided\n"
-  usage
-fi
 
-case "${bundle_type}" in
-  ("workspace") tre_get_path="api/workspace-templates" ;;
-  ("workspace_service") tre_get_path="api/workspace-service-templates" ;;
-  ("user_resource") tre_get_path="api/workspace-service-templates/${workspace_service_name}/user-resource-templates";;
-  ("shared_service") tre_get_path="api/shared-service-templates";;
-esac
+function get_template() {
+  case "${bundle_type}" in
+    ("workspace") get_result=$(tre workspace-template "$template_name" show --output json) || echo ;;
+    ("workspace_service") get_result=$(tre workspace-service-template "$template_name" show --output json) || echo ;;
+    ("user_resource") get_result=$(tre workspace-service-template "${workspace_service_name}" user-resource-template "$template_name" show --output json) || echo;;
+    ("shared_service") get_result=$(tre shared-service-template "$template_name" show --output json) || echo ;;
+  esac
+  echo "$get_result"
+}
 
-curl_settings=(--retry 7 --retry-max-time 300 --max-time 90)
 
-register_result=$(curl -i "${curl_settings[@]}" -X "POST" "${tre_url}/${tre_get_path}" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer ${access_token}" -d "${payload}" "${options}")
-get_http_code "${register_result}"
-if [[ ${http_code} == 409 ]]; then
-  echo "Template with this version already exists"
-elif [[ ${http_code} != 201 ]]; then
-  echo "Error while registering template"
-  echo "${register_result}"
-  exit 1
+get_result=$(get_template)
+if [[ -n "$(echo "$get_result" | jq -r .id)" ]]; then
+  # 'id' was returned - so we successfully got the template from the API. Now check the version
+  if [[ "$(echo "$get_result" | jq -r .version)" == "$template_version" ]]; then
+    echo "Template with this version already exists"
+    exit
+  fi
+else
+  error_code=$(echo "$get_result" | jq -r .status_code)
+  if [[  "$error_code" = "404" ]]; then
```

Comment:
We could check that error_code is not 200?  To catch any other potential errors (500?)

Reply:
Ah - actually.... it should be `!= 404`.

If it's a 404 then we didn't find the template and want to continue and install it. Otherwise, we should error.  Have pushed an update